ui/components/MesheryPerformance/index.js

```diff
@@ -335,37 +353,66 @@ class MesheryPerformanceComponent extends React.Component {
         try {
           // empty text input exception
           if (value !== '') JSON.parse(value);
-          this.setState({
-            additional_options: value,
-            jsonError: false,
-          });
+          setAdditionalOptions(value);
+          setJsonError(false);
         } catch (error) {
-          this.setState({
-            additional_options: value,
-            jsonError: true,
-          });
+          setAdditionalOptions(value);
+          setJsonError(true);
         }
       }
     }
-    this.setState({ [name]: event.target.value });
+    // to handle the controlled condition of components and
+    switch (name) {
+      case 'profileName':
+        setProfileName(event.target.value);
```

Comment:
You could have just extracted the `value` and used at all required places

Reply:
sure!